src/Repositories/CustomUserRepository.php

```diff
@@ -1,117 +1,90 @@
+     * @param $data The user data provided by auth0
      *
-     * @param array $data - user sign up data
+     * @return $user The stateful user
      *
-     * @return User
+     * @inheritdoc
      */
-    public function updateOrCreate($data): object
-    {
-        // Update or create user
-        $user = User::updateOrCreate(
-            [
-                'email' => $data['email'],
-            ],
-            [
-                'name' => $data['first_name'] . " " . $data['last_name'],
-            ]
-        );
-
-        // Check for auth0 user
-        $user_data = (new Auth0ApiRequester())->searchAuth0UserByEmail($user->email);
-        $password = isset($data['password']) ? $data['password'] : null;
-        if (empty($user_data)) {
-            $auth0_user = (new Auth0ApiRequester())->createAuth0User($user->email, $user->name, $password);
-            $user->sub = $auth0_user->user_id;
-            $user->save();
-        } else {
-
-            /**
-             * This method of account retrieval is risky as there might be multiple accounts
-             * associated with a single email address. When creating a new user though, its
-             * still the most reliable identifier we have available
-             */
-            $user->sub = $user_data[0]->user_id;
-            $user->save();
-
-            // The user has now set another password
-            if ($password) {
-                (new Auth0ApiRequester())->changePassword($user_data[0]->user_id, $password);
+    public function fromSession(
+        array $data
+    ): ?\Illuminate\Contracts\Auth\Authenticatable {
+
+        // Always create the user if it doesn't exist yet
+        $user = $this->upsertUser($data);
+
+        // This will only be true for a new signup
+        if ($user->wasRecentlyCreated) {
+
+            // Search for an existing user on Auth0
+            $user_data = (new Auth0ApiRequester())->searchAuth0UserByEmail($user->email);
+            $password = isset($data['password']) ? $data['password'] : null;
+            if (empty($user_data)) {
+                $auth0_user = (new Auth0ApiRequester())->createAuth0User($user->email, $user->name, $password);
+                $user->sub = $auth0_user->user_id;
+                $user->save();
+            } else {
+
+                /**
+                 * This method of account retrieval is risky as there might be multiple accounts
+                 * associated with a single email address. When creating a new user though, its
+                 * still the most reliable identifier we have available
+                 */
+                $user->sub = $user_data[0]->user_id;
+                $user->save();
+
+                // The user has now set another password
+                if ($password) {
+                    (new Auth0ApiRequester())->changePassword($user_data[0]->user_id, $password);
+                }
             }
         }
```

Comment:
@PhillipMwaniki heads up here that I have changed the logic to check if the user exists. Otherwise the code was making an API call to Auth0 repeatedly. 

Let's see if this helps with the bad gateway issue you were having.